function-sidecar/cmd/function-sidecar.go

```diff
@@ -95,7 +96,10 @@ func main() {
 
 	consumerConfig := makeConsumerConfig()
 	consumerConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
-	consumer, err := kafka.NewConsumer(brokers, group, []string{input}, consumerConfig)
+
+	pod := uuid.NewV4().String()
```

Comment:
Doesn't this UUID and the one from line 87 need to be the same so they can be correlated?

Reply:
I don't think so. My purpose in introducing these ids was simply to enable the autoscaler to distinguish the metrics from distinct producers and to distinguish the metrics from distinct consumers (in distinct pods). I didn't _intend_ any relationship between producer id and consumer id. Of course, we _could_ share the same UUID here, but that would be confusing given my intention.

So what kind of correlation do you anticipate between these ids? Are you perhaps thinking of a developer eyeballing the metrics?

Reply:
I _thought_ we needed to compare for a given pod, but you're right we don't. So it's better left as 2 different ides